backstage-connect/src/main/java/org/forgerock/backstage/connect/client/BackstageConnectUsageService.java

```diff
@@ -0,0 +1,337 @@
+/*
+ * The contents of this file are subject to the terms of the Common Development and
+ * Distribution License (the License). You may not use this file except in compliance with the
+ * License.
+ *
+ * You can obtain a copy of the License at legal/CDDLv1.0.txt. See the License for the
+ * specific language governing permission and limitations under the License.
+ *
+ * When distributing Covered Software, include this CDDL Header Notice in each file and include
+ * the License file at legal/CDDLv1.0.txt. If applicable, add the following below the CDDL
+ * Header, with the fields enclosed by brackets [] replaced by your own identifying
+ * information: "Portions copyright [year] [name of copyright owner]".
+ *
+ * Copyright 2017 ForgeRock AS.
+ */
+
+package org.forgerock.backstage.connect.client;
+
+import static java.util.concurrent.Executors.newScheduledThreadPool;
+import static org.forgerock.http.apache.async.AsyncHttpClientProvider.OPTION_WORKER_THREADS;
+import static org.forgerock.http.routing.Version.version;
+import static org.forgerock.util.Reject.checkNotNull;
+import static org.forgerock.util.Utils.isNullOrEmpty;
+import static org.forgerock.http.handler.HttpClientHandler.OPTION_LOADER;
+import static org.forgerock.json.JsonValue.field;
+import static org.forgerock.json.JsonValue.json;
+import static org.forgerock.json.JsonValue.object;
+import static org.forgerock.util.Utils.newThreadFactory;
+
+import java.util.Date;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+import org.forgerock.http.Client;
+import org.forgerock.http.HttpApplicationException;
+import org.forgerock.http.apache.async.AsyncHttpClientProvider;
+import org.forgerock.http.handler.HttpClientHandler;
+import org.forgerock.http.protocol.Request;
+import org.forgerock.http.protocol.Response;
+import org.forgerock.http.routing.Version;
+import org.forgerock.http.spi.Loader;
+import org.forgerock.json.JsonValue;
+import org.forgerock.json.jose.builders.JwtBuilderFactory;
+import org.forgerock.json.jose.jwt.Jwt;
+import org.forgerock.json.jose.jwt.JwtClaimsSet;
+import org.forgerock.json.jose.jwt.JwtHeader;
+import org.forgerock.util.Options;
+import org.forgerock.util.Reject;
+import org.forgerock.util.promise.ResultHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service that initiates a <em>ping</em> to ForgeRock Backstage. The following information is transmitted:
+ * <ul>
+ * <li>Product Name</li>
+ * <li>Product Version</li>
+ * <li>Unique Customer-Issued Key</li>
+ * </ul>
+ */
+public class BackstageConnectUsageService {
```

Comment:
Should we create GH issue to drop classes like this?

Reply:
I agree -- we can PR the removal after this PR is merged. As @siepkes said, I just wanted this to be historical record / code for reference for later.